kotlin/src/main/kotlin/com/example/android/learn-kotlin/java/com/example/kotlin_samples/KotlinSamples.kt

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.kotlin_samples
+
+// A top-level constant used by several snippets.
+private const val count = 42
+
+/**
+ * This function consolidates the output of all language snippets
+ * into a single string for display in the UI.
+ */
+fun getLanguageSamplesOutput(): String {
+    return """
+    --- Control Flow ---
+    ${controlFlowSnippets()}
+
+    --- Null Safety ---
+    ${nullSafetySnippets()}
+
+    --- Functions ---
+    ${functionSnippets()}
+
+    --- Higher-Order Functions ---
+    ${higherOrderFunctionSnippets()}
+    """.trimIndent()
+}
+
+private fun controlFlowSnippets(): String {
+    val output = StringBuilder()
+
+    // [START kotlin_control_flow_if_else_if]
+    val result1 = if (count == 42) {
+        "I have the answer."
```

Comment:
Same comment as below, the snippets must match exactly as they were before, so we need to add back the "println" commands.